src/h2olog/h2olog.cc

```diff
@@ -125,6 +126,44 @@ static void show_process(pid_t pid)
     infof("Attaching pid=%d (%s)", pid, cmdline);
 }
 
+static void drop_root_privilege(void)
+{
+    if (getuid() == 0) {
+        if (getgid() == 0) {
+            const char *sudo_gid = getenv("SUDO_GID");
+            if (sudo_gid == NULL) {
+                fprintf(stderr, "Error: failed to read the SUDO_GID env variable\n");
+                exit(EXIT_FAILURE);
+            }
+            errno = 0;
+            gid_t gid = (gid_t)strtol(sudo_gid, NULL, 10);
+            if (errno != 0) {
+                fprintf(stderr, "Error: overflow while parsing SUDO_GID\n");
+                exit(EXIT_FAILURE);
+            }
+            if (gid != 0 && setgid(gid) != 0) {
```

Comment:
I think we'd either remove `gid != 0` or call it an error.

Though it depends on how we define what "dropping root privileges" mean. If we are to define the act as switching to SUDO_UID / SUDO_GID, then simply removing this condition is fine. If we are to define the act as "dropping root privileges" in a strict manner, then we should call it an error.

Reply:
Agreed that we need to decide on the definition, and I'm good with simply: "switching to SUDO_UID / SUDO_GID" as you mentioned. I see your point about needing to be more unforgiving if we had chosen a stricter definition. Thanks!